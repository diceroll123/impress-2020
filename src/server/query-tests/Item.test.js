const gql = require("graphql-tag");
const { query, getDbCalls } = require("./setup.js");

describe("Item", () => {
  it("loads metadata", async () => {
    const res = await query({
      query: gql`
        query {
          items(ids: ["38913", "38911", "38912", "55788", "77530", "78104"]) {
            id
            name
            description
            thumbnailUrl
            rarityIndex
            isNc
            manualSpecialColor {
              id
              name
            }
            explicitlyBodySpecific
          }
        }
      `,
    });

    expect(res).toHaveNoErrors();
    expect(res.data).toMatchSnapshot();
    expect(getDbCalls()).toMatchInlineSnapshot(`
      Array [
        Array [
          "SELECT * FROM item_translations WHERE item_id IN (?,?,?,?,?,?) AND locale = \\"en\\"",
          Array [
            "38913",
            "38911",
            "38912",
            "55788",
            "77530",
            "78104",
          ],
        ],
        Array [
          "SELECT * FROM items WHERE id IN (?,?,?,?,?,?)",
          Array [
            "38913",
            "38911",
            "38912",
            "55788",
            "77530",
            "78104",
          ],
        ],
        Array [
          "SELECT * FROM color_translations
             WHERE color_id IN (?) AND locale = \\"en\\"",
          Array [
            "44",
          ],
        ],
      ]
    `);
  });

  it("loads appearance data", async () => {
    const res = await query({
      query: gql`
        query {
          items(ids: ["38912", "38911", "37375"]) {
            id
            name

            appearanceOn(speciesId: "54", colorId: "75") {
              layers {
                id
                remoteId
                imageUrl(size: SIZE_600)
                svgUrl
                zone {
                  id
                  depth
                  label
                }
              }

              restrictedZones {
                id
              }
            }
          }
        }
      `,
    });

    expect(res).toHaveNoErrors();
    expect(res.data).toMatchSnapshot();
    expect(getDbCalls()).toMatchInlineSnapshot(`
      Array [
        Array [
          "SELECT * FROM item_translations WHERE item_id IN (?,?,?) AND locale = \\"en\\"",
          Array [
            "38912",
            "38911",
            "37375",
          ],
        ],
        Array [
          "SELECT * FROM pet_types WHERE (species_id = ? AND color_id = ?)",
          Array [
            "54",
            "75",
          ],
        ],
        Array [
          "SELECT sa.*, rel.parent_id FROM swf_assets sa
             INNER JOIN parents_swf_assets rel ON
               rel.parent_type = \\"Item\\" AND
               rel.swf_asset_id = sa.id
             WHERE (rel.parent_id = ? AND (sa.body_id = ? OR sa.body_id = 0)) OR (rel.parent_id = ? AND (sa.body_id = ? OR sa.body_id = 0)) OR (rel.parent_id = ? AND (sa.body_id = ? OR sa.body_id = 0))",
          Array [
            "38912",
            "180",
            "38911",
            "180",
            "37375",
            "180",
          ],
        ],
        Array [
          "SELECT * FROM items WHERE id IN (?,?,?)",
          Array [
            "38912",
            "38911",
            "37375",
          ],
        ],
        Array [
          "SELECT * FROM zones WHERE id IN (?,?,?)",
          Array [
            "26",
            "40",
            "3",
          ],
        ],
        Array [
          "SELECT * FROM zone_translations WHERE zone_id IN (?,?,?) AND locale = \\"en\\"",
          Array [
            "26",
            "40",
            "3",
          ],
        ],
      ]
    `);
  });

  it("skips appearance data for audio assets", async () => {
    const res = await query({
      query: gql`
        query {
          items(ids: ["42829"]) {
            id
            name

            appearanceOn(speciesId: "54", colorId: "75") {
              layers {
                id
                imageUrl(size: SIZE_600)
                svgUrl
                zone {
                  id
                  depth
                  label
                }
              }

              restrictedZones {
                id
              }
            }
          }
        }
      `,
    });

    expect(res).toHaveNoErrors();
    expect(res.data).toMatchSnapshot();
    expect(getDbCalls()).toMatchInlineSnapshot(`
      Array [
        Array [
          "SELECT * FROM item_translations WHERE item_id IN (?) AND locale = \\"en\\"",
          Array [
            "42829",
          ],
        ],
        Array [
          "SELECT * FROM pet_types WHERE (species_id = ? AND color_id = ?)",
          Array [
            "54",
            "75",
          ],
        ],
        Array [
          "SELECT sa.*, rel.parent_id FROM swf_assets sa
             INNER JOIN parents_swf_assets rel ON
               rel.parent_type = \\"Item\\" AND
               rel.swf_asset_id = sa.id
             WHERE (rel.parent_id = ? AND (sa.body_id = ? OR sa.body_id = 0))",
          Array [
            "42829",
            "180",
          ],
        ],
        Array [
          "SELECT * FROM items WHERE id IN (?)",
          Array [
            "42829",
          ],
        ],
      ]
    `);
  });
});
