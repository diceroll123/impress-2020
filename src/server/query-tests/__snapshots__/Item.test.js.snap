// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item loads appearance data 1`] = `
Object {
  "items": Array [
    Object {
      "appearanceOn": Object {
        "layers": Array [
          Object {
            "id": "37128",
            "imageUrl": "https://impress-asset-images.s3.amazonaws.com/object/000/000/014/14856/600x600.png?v2-1587653266000",
            "remoteId": "14856",
            "svgUrl": null,
            "zone": Object {
              "depth": 30,
              "id": "26",
              "label": "Jacket",
            },
          },
        ],
        "restrictedZones": Array [
          Object {
            "id": "20",
          },
          Object {
            "id": "22",
          },
        ],
      },
      "id": "38912",
      "name": "Zafara Agent Robe",
    },
    Object {
      "appearanceOn": Object {
        "layers": Array [
          Object {
            "id": "37129",
            "imageUrl": "https://impress-asset-images.s3.amazonaws.com/object/000/000/014/14857/600x600.png?v2-0",
            "remoteId": "14857",
            "svgUrl": null,
            "zone": Object {
              "depth": 44,
              "id": "40",
              "label": "Hat",
            },
          },
        ],
        "restrictedZones": Array [
          Object {
            "id": "37",
          },
          Object {
            "id": "38",
          },
        ],
      },
      "id": "38911",
      "name": "Zafara Agent Hood",
    },
    Object {
      "appearanceOn": Object {
        "layers": Array [
          Object {
            "id": "30203",
            "imageUrl": "https://impress-asset-images.s3.amazonaws.com/object/000/000/006/6829/600x600.png?v2-1598519675000",
            "remoteId": "6829",
            "svgUrl": null,
            "zone": Object {
              "depth": 3,
              "id": "3",
              "label": "Background",
            },
          },
        ],
        "restrictedZones": Array [],
      },
      "id": "37375",
      "name": "Moon and Stars Background",
    },
  ],
}
`;

exports[`Item loads metadata 1`] = `
Object {
  "items": Array [
    Object {
      "description": "Dont leave any trace that you were there with these gloves.",
      "explicitlyBodySpecific": false,
      "id": "38913",
      "isNc": false,
      "manualSpecialColor": null,
      "name": "Zafara Agent Gloves",
      "rarityIndex": 88,
      "thumbnailUrl": "http://images.neopets.com/items/clo_zafara_agent_gloves.gif",
    },
    Object {
      "description": "Hide your face and hair so no one can recognise you.",
      "explicitlyBodySpecific": false,
      "id": "38911",
      "isNc": false,
      "manualSpecialColor": null,
      "name": "Zafara Agent Hood",
      "rarityIndex": 92,
      "thumbnailUrl": "http://images.neopets.com/items/clo_zafara_agent_hood.gif",
    },
    Object {
      "description": "This robe is great for being stealthy.",
      "explicitlyBodySpecific": false,
      "id": "38912",
      "isNc": false,
      "manualSpecialColor": null,
      "name": "Zafara Agent Robe",
      "rarityIndex": 90,
      "thumbnailUrl": "http://images.neopets.com/items/clo_zafara_agent_robe.gif",
    },
    Object {
      "description": "Maybe youll be discovered by some Neopets from the future and thawed out!",
      "explicitlyBodySpecific": true,
      "id": "55788",
      "isNc": true,
      "manualSpecialColor": null,
      "name": "Encased in Ice",
      "rarityIndex": 500,
      "thumbnailUrl": "http://images.neopets.com/items/mall_petinice.gif",
    },
    Object {
      "description": "Made with the finest jewels of the sea!",
      "explicitlyBodySpecific": false,
      "id": "77530",
      "isNc": true,
      "manualSpecialColor": Object {
        "id": "44",
        "name": "Maraquan",
      },
      "name": "Maraquan Sea Blue Gown",
      "rarityIndex": 500,
      "thumbnailUrl": "http://images.neopets.com/items/mall_clo_marabluegown.gif",
    },
    Object {
      "description": "You truly are the number one fan of Altador Cup, and your room reflects this!",
      "explicitlyBodySpecific": false,
      "id": "78104",
      "isNc": true,
      "manualSpecialColor": null,
      "name": "#1 Fan Room Background",
      "rarityIndex": 500,
      "thumbnailUrl": "http://images.neopets.com/items/mall_bg_numberonefanbg.gif",
    },
  ],
}
`;

exports[`Item skips appearance data for audio assets 1`] = `
Object {
  "items": Array [
    Object {
      "appearanceOn": Object {
        "layers": Array [],
        "restrictedZones": Array [],
      },
      "id": "42829",
      "name": "Time Tunnel Music Track",
    },
  ],
}
`;
